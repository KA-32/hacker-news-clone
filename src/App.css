.main {
  text-align: center;
  width: 100%;
  position: relative;
  margin-bottom: 5em;
}

.loader-wrapper {
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  top: 0;
}

.loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid orange;
  width: 80px;
  height: 80px;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
  margin:10em auto 0 auto;
}

@-webkit-keyframes spin {
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.recharts-wrapper {
  margin: 1em auto 0 auto;
  border-top: 6px solid orange;
  border-bottom: 6px solid orange;
  padding: 1em 0;
}

.recharts-legend-wrapper {
  visibility: hidden;
}
