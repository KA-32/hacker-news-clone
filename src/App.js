import React from "react";

import NewsFeed from "./components/NewsFeed/NewsFeed";

import "./App.css";

function App() {
  return (
    <section className="main">
      <NewsFeed />
    </section>
  );
}

export default App;
