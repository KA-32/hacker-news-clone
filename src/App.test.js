import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("App component", () => {
  test("App renders correct table headers", () => {
    const { getByText } = render(<App data={{}} isLineChartVisible={true} />);
    const commentsElement = getByText(/Comments/i);
    const voteCountElement = getByText(/Vote Count/i);
    const upvoteElement = getByText(/Upvote/i);
    const detailsElement = getByText(/News Details/i);

    expect(commentsElement).toBeInTheDocument();
    expect(voteCountElement).toBeInTheDocument();
    expect(upvoteElement).toBeInTheDocument();
    expect(detailsElement).toBeInTheDocument();
  });

  test("App to match snapshot", () => {
    const app = render(<App data={{}} isLineChartVisible={true} />);
    expect(app).toMatchSnapshot();
  });
});
