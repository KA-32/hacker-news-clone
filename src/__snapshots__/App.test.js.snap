// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main"
      >
        <section
          class="main-container"
          role="main"
        >
          <table
            class="news-feed-table"
          >
            <thead>
              <tr
                class="news-feed-header-row"
              >
                <th
                  class="news-feed-header-item fixed-width center"
                >
                  Comments
                </th>
                <th
                  class="news-feed-header-item fixed-width center"
                >
                  Vote Count
                </th>
                <th
                  class="news-feed-header-item fixed-width center"
                >
                  Upvote
                </th>
                <th
                  class="news-feed-header-item left-align"
                >
                  News Details
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <div
            class="btn-wrapper"
          >
            <button
              class="prev"
              disabled=""
            >
              Previous
            </button>
            <button
              class="next"
            >
              Next
            </button>
          </div>
        </section>
        <div
          class="loader-wrapper"
        >
          <div
            class="loader"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main"
    >
      <section
        class="main-container"
        role="main"
      >
        <table
          class="news-feed-table"
        >
          <thead>
            <tr
              class="news-feed-header-row"
            >
              <th
                class="news-feed-header-item fixed-width center"
              >
                Comments
              </th>
              <th
                class="news-feed-header-item fixed-width center"
              >
                Vote Count
              </th>
              <th
                class="news-feed-header-item fixed-width center"
              >
                Upvote
              </th>
              <th
                class="news-feed-header-item left-align"
              >
                News Details
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
        <div
          class="btn-wrapper"
        >
          <button
            class="prev"
            disabled=""
          >
            Previous
          </button>
          <button
            class="next"
          >
            Next
          </button>
        </div>
      </section>
      <div
        class="loader-wrapper"
      >
        <div
          class="loader"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
